cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go

```diff
@@ -280,6 +281,61 @@ var _ = Describe("[sriov] NUMA node alignment", Ordered, func() {
 			g.Expect(actualPod.Status.Message).To(ContainSubstring("Resources cannot be allocated with Topology locality"))
 		}, 30*time.Second, 1*time.Second).Should(Succeed())
 	})
+
+	It("Utilize all available VFs then create a pod with guaranteed CPU and excludeTopology set to True", func() {
+		podDefinition := pods.DefinePod(sriovnamespaces.Test)
+		podDefinition = pods.RedefineWithGuaranteedQoS(podDefinition, "2", "500Mi")
+
+		numVFs := 4
+		By("Utilize all available VFs")
+		expectedRunningPod := podDefinition.DeepCopy()
+		useAllVFsNetworkSpec := []string{}
+		for i := 0; i < numVFs; i++ {
+			useAllVFsNetworkSpec = append(useAllVFsNetworkSpec, "test-numa-0-nic1-exclude-topology-true-network")
+		}
+		expectedRunningPod = pods.RedefinePodWithNetwork(expectedRunningPod, strings.Join(useAllVFsNetworkSpec, ","))
+
+		expectedRunningPod, err := client.Client.Pods(sriovnamespaces.Test).
+			Create(context.Background(), expectedRunningPod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			actualPod, err := client.Client.Pods(sriovnamespaces.Test).Get(context.Background(), expectedRunningPod.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(actualPod.Status.Phase).To(Equal(corev1.PodRunning))
+			g.Expect(actualPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+		}, 30*time.Second, 1*time.Second).Should(Succeed())
+
+		By("Create a failing pod that uses a VF")
+		failingPod := podDefinition.DeepCopy()
+		failingPod = pods.RedefinePodWithNetwork(failingPod, "test-numa-0-nic1-exclude-topology-true-network")
+		failingPod, err = client.Client.Pods(sriovnamespaces.Test).
+			Create(context.Background(), failingPod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(pods.GetStringEventsForPodFn(client.Client, failingPod), 30*time.Second, 1*time.Second).
+			Should(ContainSubstring("Insufficient openshift.io/testNuma0NIC1ExcludeTopoplogyTrue"))
+
+		By("Release all VFs by deleting the running pod")
+		err = client.Client.Pods(sriovnamespaces.Test).
+			Delete(context.Background(), expectedRunningPod.Name, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Create a pod to verify VFs are usable")
```

Comment:
should the failing pod be running too here?

Reply:
good point. I tested locally and it actually gets started. I can simplify the test by removing a pod from the game